lib/acme/client.rb

```diff
@@ -66,14 +66,19 @@ def connection
     end
   end
 
-  def challenge_from_hash(attributes)
-    case attributes.fetch('type')
+  def challenge_from_hash(arguments)
+    arguments.default_proc = lambda { |_, key| raise "Missing #{key} key" }
```

Comment:
Sorry I don't like this much at all:

* It's raising a `RuntimeError`, if one would want to handle it, one would need to interpret the associated message. This arguably worse than the standard `KeyError`.
* `Hash#default_proc=` is modifying the object, so this modifies the callers object and can have unexpected and surprising side effects in case the caller still uses the passed value elsewhere afterwards.